plugins/logitech-hidpp/fu-logitech-hidpp-device.c

```diff
@@ -351,13 +361,42 @@ fu_logitech_hidpp_device_feature_get_idx(FuLogitechHidPpDevice *self, guint16 fe
 	return 0x00;
 }
 
+static gboolean
+fu_logitech_hidpp_device_create_radio_child(FuLogitechHidPpDevice *self,
+					    guint8 entity,
+					    guint16 build,
+					    GError **error)
+{
+	FuLogitechHidPpDevicePrivate *priv = GET_PRIVATE(self);
+	FuContext *ctx = fu_device_get_context(FU_DEVICE(self));
+	g_autofree gchar *instance_id = NULL;
+	g_autofree gchar *radio_version = NULL;
+	g_autoptr(FuLogitechHidPpRadio) radio = NULL;
+
+	radio_version = g_strdup_printf("0x%.4x", build);
+	radio = fu_logitech_hidpp_radio_new(ctx, entity);
+	fu_device_set_physical_id(FU_DEVICE(radio), fu_device_get_physical_id(FU_DEVICE(self)));
+
+	fu_device_set_logical_id(FU_DEVICE(radio), priv->model_id);
```

Comment:
don't you have a case here that `priv->model_id` could potentially be `NULL` and make a bad instance ID for a "new" device not yet supported?  To be safe, I think you should guard against this when creating children.

Reply:
Yes, good point. It works now, but without a check we're causing our future selves some head scratching. I've added an explicit check now.